File: src/macos/AzPin/Models/CachedToken.swift
import SwiftData
import Foundation

@Model
final class CachedToken {
    var subscriptionId: String
    var tenantId: String
    var accessToken: String
    var expiresOn: Date

    init(subscriptionId: String, tenantId: String, accessToken: String, expiresOn: Date) {
        self.subscriptionId = subscriptionId
        self.tenantId = tenantId
        self.accessToken = accessToken
        self.expiresOn = expiresOn
    }
}

File: src/windows/AzPin.Windows.Core/Models/Entities/CachedToken.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace AzPin.Windows.Models.Entities;

[Index(nameof(SubscriptionId), IsUnique = true)]
public class CachedToken
{
    [Key]
    public int LocalId { get; set; }
    public string SubscriptionId { get; set; } = string.Empty;
    public string TenantId { get; set; } = string.Empty;
    public string AccessToken { get; set; } = string.Empty;
    public DateTime ExpiresOn { get; set; }
}
